Android/app/src/go/intra/split/retrier.go

```diff
@@ -211,12 +209,12 @@ func (r *retrier) retry(buf []byte) (n int, err error) {
 }
 
 func (r *retrier) CloseRead() error {
-	if !r.readClosed() {
-		close(r.readCloseFlag)
+	if r.readCloseFlag.CompareAndSwap(false, true) {
+		r.mutex.Lock()
+		defer r.mutex.Unlock()
+		return r.conn.CloseRead()
```

Comment:
Please preserve the current behavior of forwarding multiple `CloseRead` calls to `r.conn`. This is useful when the first `r.conn.CloseRead()` returns an `error`, and the caller tries to call `CloseRead()` again later, we should forward the second call to `r.conn`. So we may simply need a `r.readCloseFlag.Store(true)` here.

Reply:
thanks! done.